xocto/ranges.py

```diff
@@ -1151,3 +1182,19 @@ def iterate_over_months(
 
         yield FiniteDatetimeRange(start_at, this_end)
         start_at = next_start
+
+
+# Subscripted generics may not be used with isinstance directly.
+# TODO: A TypeGuard would be nicer, once we drop Python 3.9.
+def _is_datetime_range(value: Any) -> bool:
+    if value.start is not None and not isinstance(value.start, datetime.datetime):
+        return False
+    if value.end is not None and not isinstance(value.end, datetime.datetime):
+        return False
+    return True
```

Comment:
That’s a lot of `not`s and `False`s, I might find it easier to understand by simply putting everything together as a positive expression:

```suggestion
    return (
        (value.start is None or isinstance(value.start, datetime.datetime))
        and (value.end is None or isinstance(value.end, datetime.datetime))
    )
```

Reply:
Sure, that's probably a little more readable - updated.